tests/integration/test_git_import/test.py

```diff
@@ -0,0 +1,186 @@
+            commit_message String,
+            commit_files_added UInt32,
+            commit_files_deleted UInt32,
+            commit_files_renamed UInt32,
+            commit_files_modified UInt32,
+            commit_lines_added UInt32,
+            commit_lines_deleted UInt32,
+            commit_hunks_added UInt32,
+            commit_hunks_removed UInt32,
+            commit_hunks_changed UInt32
+        ) ENGINE = MergeTree ORDER BY time
+        """
+    )
+
+    node.query(
+        f"""
+        CREATE TABLE line_changes
+        (
+            sign Int8,
+            line_number_old UInt32,
+            line_number_new UInt32,
+            hunk_num UInt32,
+            hunk_start_line_number_old UInt32,
+            hunk_start_line_number_new UInt32,
+            hunk_lines_added UInt32,
+            hunk_lines_deleted UInt32,
+            hunk_context LowCardinality(String),
+            line LowCardinality(String),
+            indent UInt8,
+            line_type Enum('Empty' = 0, 'Comment' = 1, 'Punct' = 2, 'Code' = 3),
+
+            prev_commit_hash String,
+            prev_author LowCardinality(String),
+            prev_time DateTime,
+
+            file_change_type Enum('Add' = 1, 'Delete' = 2, 'Modify' = 3, 'Rename' = 4, 'Copy' = 5, 'Type' = 6),
+            path LowCardinality(String),
+            old_path LowCardinality(String),
+            file_extension LowCardinality(String),
+            file_lines_added UInt32,
+            file_lines_deleted UInt32,
+            file_hunks_added UInt32,
+            file_hunks_removed UInt32,
+            file_hunks_changed UInt32,
+
+            commit_hash String,
+            author LowCardinality(String),
+            time DateTime,
+            commit_message String,
+            commit_files_added UInt32,
+            commit_files_deleted UInt32,
+            commit_files_renamed UInt32,
+            commit_files_modified UInt32,
+            commit_lines_added UInt32,
+            commit_lines_deleted UInt32,
+            commit_hunks_added UInt32,
+            commit_hunks_removed UInt32,
+            commit_hunks_changed UInt32
+        ) ENGINE = MergeTree ORDER BY time
+        """
+    )
+
+
+def insert_into_tables(dir):
+    command = f'/usr/bin/clickhouse client --query="INSERT INTO commits FORMAT TSV" < {dir}/commits.tsv'
+    node.exec_in_container(["bash", "-c", command])
+
+    command = f'/usr/bin/clickhouse client --query="INSERT INTO file_changes FORMAT TSV" < {dir}/file_changes.tsv'
+    node.exec_in_container(["bash", "-c", command])
+
+    command = f'/usr/bin/clickhouse client --query="INSERT INTO line_changes FORMAT TSV" < {dir}/line_changes.tsv'
+    node.exec_in_container(["bash", "-c", command])
+
+
+def drop_tables():
+    node.query(f"DROP TABLE commits")
+    node.query(f"DROP TABLE file_changes")
+    node.query(f"DROP TABLE line_changes")
+
+
+def test_git_import():
+    repo = "https://github.com/githubtraining/hellogitworld.git"
```

Comment:
https://github.com/githubtraining/hellogitworld is a just a random repository small enough to use in a test.
Earlier we used https://github.com/ClickHouse/clickhouse-odbc.git here (see [commit](https://github.com/ClickHouse/ClickHouse/pull/73866/commits/a05702f8f4616ed258775c735f77dba025a99a69)).